qemu/tests/eject_media.py

```diff
@@ -24,90 +24,78 @@ def run(test, params, env):
     :param env: Dictionary with test environment
     """
 
-    qemu_binary = utils_misc.get_qemu_binary(params)
-    if not utils_misc.qemu_has_option("qmp", qemu_binary):
-        logging.warn("qemu does not support qmp. Human monitor will be used.")
-    qmp_used = False
     vm = env.get_vm(params["main_vm"])
     vm.verify_alive()
 
     session = vm.wait_for_login(timeout=int(params.get("login_timeout", 360)))
     logging.info("Wait until device is ready")
     time.sleep(10)
-    if vm.monitor.protocol == "qmp":
-        qmp_used = True
+
+    @utils_test.qemu.fail_test_on_exception([virt_vm.VMDeviceError])
```

Comment:
We already have this one, please have a look.

http://avocado-framework.readthedocs.io/en/59.0/api/test/avocado.html#avocado.fail_on